File: services/alert-engine/src/cache.ts
import { createClient } from "redis";
import { config } from "./config";

export const redis = createClient({ url: config.REDIS_URL });

export async function connectCache(): Promise<void> {
  if (!redis.isOpen) {
    await redis.connect();
  }
}

export async function checkCache(): Promise<void> {
  await connectCache();
  await redis.ping();
}

export async function closeCache(): Promise<void> {
  if (redis.isOpen) {
    await redis.quit();
  }
}

File: services/market-data/app/cache.py
from __future__ import annotations

from redis.asyncio import Redis

from app.config import settings

redis_client = Redis.from_url(settings.redis_url, decode_responses=True)


async def connect_cache() -> None:
    await redis_client.ping()


async def check_cache() -> None:
    await redis_client.ping()


async def close_cache() -> None:
    await redis_client.close()
